packages/dialtone-vue2/recipes/item_layout/contact_info/contact_info.vue

```diff
@@ -26,23 +26,46 @@
             :full-name="avatar.fullName"
             :image-src="avatar.src"
             image-alt=""
-            :icon-name="avatarIcon"
-            :overlay-icon="avatar.icon"
             :overlay-text="avatar.text"
             :avatar-class="[{ 'd-mln24': index > 0, 'd-bc-brand': !!avatar.halo }]"
-          />
+          >
+            <template
+              v-if="avatarIcon"
+              #icon="{ iconSize }"
+            >
+              <component
+                :is="avatarIcon"
+                :size="iconSize"
+              />
+            </template>
+            <template
+              v-if="avatar.icon"
+              #overlayIcon
+            >
+              <component :is="avatar.icon" />
```

Comment:
Same here, why we're expecting icon component instead of a slot?

Reply:
This one is a bit weird because it's not just a single avatar it's a list of avatars passed in via an array prop. Discussing with Nina the best way to solve this.

Reply:
I tested this on a test project and the tree shaking works, the bundle only includes the icon that I am passing as a prop. Maybe it's not the most consistent with how the rest of our components work, but also this is used in only one place in the product and it would be a huge change to convert the whole list to a slot. What do you think about going with this solution for now and creating a tech debt ticket to improve it later?

Reply:
Fine by me, as long as tree shaking works